src/test/java/reactor/kafka/sender/internals/MockSenderTest.java

```diff
@@ -150,7 +150,6 @@ public void sendNoResponseFailure() {
         StepVerifier.create(sender.createOutbound().send(outgoing.producerRecords()).then())
                     .expectError(InvalidTopicException.class)
                     .verify(Duration.ofMillis(DEFAULT_TEST_TIMEOUT));
-        assertEquals(maxInflight, outgoing.onNextCount.get());
```

Comment:
the assertion wasn't correct (breaks due to prefetching). We have `producer.enableInFlightCheck();` a few lines earlier that will ensure that `maxInflight` is correct.